OsmAnd/src/net/osmand/plus/plugins/externalsensors/dialogs/SelectExternalDeviceFragment.java

```diff
@@ -0,0 +1,227 @@
+			int widgetFieldTypeOrdinal = arguments.getInt(WIDGET_TYPE_KEY);
+			widgetDataFieldType = SensorWidgetDataFieldType.values()[widgetFieldTypeOrdinal];
+			selectedDeviceId = arguments.getString(SELECTED_DEVICE_ID_KEY);
+		}
+	}
+
+	@Override
+	protected void setupUI(@NonNull View view) {
+		super.setupUI(view);
+		emptyView = view.findViewById(R.id.empty_view);
+		contentView = view.findViewById(R.id.devices_content);
+		devicesList = view.findViewById(R.id.connected_devices_list);
+		appBar = view.findViewById(R.id.appbar);
+		noBluetoothCard = view.findViewById(R.id.no_bluetooth_card);
+
+		setupPairSensorButton(view.findViewById(R.id.pair_btn_empty));
+		setupPairSensorButton(view.findViewById(R.id.pair_btn_additional));
+		setupOpenBtSettingsButton(view.findViewById(R.id.bt_settings_button_container));
+		devicesListAdapter = new DevicesForWidgetAdapter(app, nightMode, this);
+		devicesListAdapter.setDeviceId(selectedDeviceId);
+		devicesList.setAdapter(devicesListAdapter);
+	}
+
+	private void setupPairSensorButton(@NonNull View view) {
+		View dismissButton = view.findViewById(R.id.dismiss_button);
+		int buttonTextId = R.string.ant_plus_pair_new_sensor;
+		ViewGroup.LayoutParams layoutParams = dismissButton.getLayoutParams();
+		UiUtilities.setupDialogButton(nightMode, dismissButton, UiUtilities.DialogButtonType.SECONDARY, buttonTextId);
+		layoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT;
+		dismissButton.setLayoutParams(layoutParams);
+		view.requestLayout();
+		dismissButton.setOnClickListener(v -> showPairNewSensorBottomSheet());
+		AndroidUiHelper.updateVisibility(dismissButton, true);
+	}
+
+	private void setupOpenBtSettingsButton(@NonNull View view) {
+		View dismissButton = view.findViewById(R.id.dismiss_button);
+		int buttonTextId = R.string.ant_plus_open_settings;
+		ViewGroup.LayoutParams layoutParams = dismissButton.getLayoutParams();
+		UiUtilities.setupDialogButton(nightMode, dismissButton, UiUtilities.DialogButtonType.SECONDARY, buttonTextId);
+		layoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT;
+		dismissButton.setLayoutParams(layoutParams);
+		view.requestLayout();
+		dismissButton.setOnClickListener(v -> {
+			Intent intentOpenBluetoothSettings = new Intent();
+			intentOpenBluetoothSettings.setAction(Settings.ACTION_BLUETOOTH_SETTINGS);
+			startActivity(intentOpenBluetoothSettings);
+		});
+		AndroidUiHelper.updateVisibility(dismissButton, true);
+	}
+
+	private void showPairNewSensorBottomSheet() {
+		PairNewDeviceBottomSheet.showInstance(requireActivity().getSupportFragmentManager());
+	}
+
+	@Override
+	protected void setupToolbar(@NonNull View view) {
+		super.setupToolbar(view);
+		Toolbar toolbar = view.findViewById(R.id.toolbar);
+		toolbar.setOnMenuItemClickListener(item -> {
+			if (item.getItemId() == R.id.action_add) {
+				showPairNewSensorBottomSheet();
+			}
+			return false;
+		});
+	}
+
+	@Override
+	public void onDestroyView() {
+		super.onDestroyView();
+		contentView = null;
```

Comment:
Is it necessary?

Reply:
there are some topics like [https://stackoverflow.com/questions/58080268/keeping-a-reference-to-a-view-in-a-fragment-causes-memory-leaks/58248851](here). keeping links to found view may produce memory leaks. I'm not sure it is our case, but i try to null view links